release-docker.sh

```diff
@@ -1,14 +1,25 @@
-#!/bin/bash
+#!/bin/sh
 
-tag=$(git describe --exact-match 2>/dev/null)
-if [[ $? != 0 ]]; then
-	tag=latest
+set -eu
+
+if ! tag="$(git describe --exact-match 2> "/dev/null")"; then
+	tag="latest"
 fi
 
-docker build -t mvdan/shfmt:$tag -f cmd/shfmt/Dockerfile .
+docker build --tag "mvdan/shfmt:${tag}" \
+             --file "cmd/shfmt/Dockerfile" \
+             "./"
+
+docker build --tag "mvdan/shfmt:${tag}-alpine" \
+             --target "alpine" \
+             --file "cmd/shfmt/Dockerfile" \
+             "."
 
-if [[ -n $DOCKER_PASSWORD ]]; then
-	echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
+if [ -n "${DOCKER_PASSWORD:-}" ]; then
+	echo "${DOCKER_PASSWORD}" | docker login --password-stdin \
+	                                         --username "${DOCKER_USERNAME}"
 fi
 
-docker push mvdan/shfmt:$tag
+docker push "mvdan/shfmt:${tag}" "mvdan/shfmt:${tag}-alpine"
+
+exit 0
```

Comment:
I've never seen this pattern - why?

Reply:
I think it's in the google bash rules.

It basically states, always have a clean and reliable  exit point. For example, if someone manages to append code to your script due some leak in the system, the script would not execute this rogue code. Also `set -eu` ensures if you reach this point, exiting is safe.